src/plugins/data/public/query/state_sync/connect_to_query_state.ts

```diff
@@ -31,12 +31,101 @@
 import { Subscription } from 'rxjs';
 import { filter, map } from 'rxjs/operators';
 import _ from 'lodash';
-import { BaseStateContainer } from '../../../../opensearch_dashboards_utils/public';
+import {
+  BaseStateContainer,
+  IOsdUrlStateStorage,
+} from '../../../../opensearch_dashboards_utils/public';
 import { QuerySetup, QueryStart } from '../query_service';
 import { QueryState, QueryStateChange } from './types';
 import { FilterStateStore, COMPARE_ALL_OPTIONS, compareFilters } from '../../../common';
 import { validateTimeRange } from '../timefilter';
 
+export const useQueryStateWithNoContainer = (
```

Comment:
Can we call it `connectStorageToQueryState`? Feels more inline with the other function

Reply:
Agree and changed.